src/index.ts

```diff
@@ -52,11 +53,16 @@ export const logger = (options: LoggerOptions = {}) => {
 
   if (!enabled) return app;
 
-  const getTimestamp = () => {
-    const now = new Date();
-    return now.toLocaleString();
+  const defaultGetTimestamp = () => {
+    return new Date().toLocaleString();
   };
 
+  // use the `withTimestamp` option if it's a function
+  const getTimestamp =
+    typeof options.withTimestamp === "function"
+      ? options.withTimestamp
+      : defaultGetTimestamp;
```

Comment:
Can you inline this function to `() => new Date().toLocaleString()` and remove `defaultGetTimestamp`. Would look much cleaner